### PR TITLE
Create ci_actions.yml

### PR DESCRIPTION
**Pull request by jorgepiloto**
Implements a CI GitHub action for code quality checking.

### CODE REVIEW

**Comment by jorgepiloto** [comment]
Seems to be working properly. By the way, it needs a rebase :+1:

**Comment by jorgepiloto** [comment]
The flake8 tool suggests to remove the `setup_orekit_curdir` from `orekit_orbit.py`. However, if this line is removed, the `pytest tests/` does not execute properly... This is weird because should be fixed by the custom `setup_orekit_env` function in `utils.py` :thinking:

**Comment by jorgepiloto** [comment]
Cool! The `orekit/tests` are now being executed by the CI action. I will try to finish this tomorrow so it is finally ready for a review :rocket:

**Review by astrojuanlu — COMMENTED** [review]
Looks good so far! Ping me when this is ready for review :)

**Comment by jorgepiloto** [comment]
> The flake8 tool suggests to remove the `setup_orekit_curdir` from `orekit_orbit.py`. However, if this line is removed, the `pytest tests/` does not execute properly... This is weird because should be fixed by the custom `setup_orekit_env` function in `utils.py` thinking

It seems that `orekit-data.zip` holds all the `org.` routines. Hence, our custom function `setup_orekit_env()` should be defined before all the `org.orekit` and `org.hipparchus` import statements.

Hoever, this new order makes `flake8` to raise the `E402` error, see:

```bash
E402 module level import not at top of file
```

How could we proceed, @astrojuanlu? Should we ignore this error in the `ci_actions.yml` under the `extra-flake8-options` variable?

**Comment by astrojuanlu** [comment]
⚠️ HISTORY HAS BEEN REWRITTEN ⚠️ This will need a rebase

**Comment by astrojuanlu** [comment]
About your comment above @jorgepiloto , I suggest we do this instead: https://flake8.pycqa.org/en/3.1.1/user/ignoring-errors.html#in-line-ignoring-errors

**Comment by jorgepiloto** [comment]
Sure! I did not know about ignoring inline errors :open_mouth:

File: orekit/src/orekit_orbit.py [hunk 1]
```diff
@@ -1,23 +1,19 @@
...
 from astropy import units as u
-from orekit.pyhelpers import setup_orekit_curdir
-from org.hipparchus.geometry.euclidean.threed import Vector3D
-from org.orekit.frames import FramesFactory
-from org.orekit.orbits import KeplerianOrbit, OrbitType, PositionAngle
-from org.orekit.propagation.analytical import KeplerianPropagator
-from org.orekit.time import AbsoluteDate
-from org.orekit.utils import Constants, PVCoordinates
+# Start orekit virtual machine, load data and make flake8 to ignore this line
 from utils import setup_orekit_env
```
Comment (astrojuanlu) [inline orekit/src/orekit_orbit.py:6]: On second thought: do you see any disadvantage on doing

```suggestion
from utils import vm
```

and make `utils.py` call

```
vm = setup_orekit_env()
```

at the end? The advantages of this approach are:

1. Python modules act like a singleton, therefore we have a single `vm` for everyone importing the module
2. We avoid the `# noqa: E402`

Comment (astrojuanlu) [reply orekit/src/orekit_orbit.py:6]: (It `utils.py` contains more stuff, perhaps a dedicated module could be created only for this)

Comment (jorgepiloto) [reply orekit/src/orekit_orbit.py:6]: Oh! I see... I completely agree with this approach, let me check if it works as expected.

Comment (jorgepiloto) [reply orekit/src/orekit_orbit.py:6]: Previous approach simplifies a lot, but `noqa: F401` is still required. Nevertheless, code became more clean, let me push the changes for you to check.

Comment (jorgepiloto) [reply orekit/src/orekit_orbit.py]: Notice that the `# isort: split` is required as `from utils import vm` statement needs to be before `from org.` ones. However, `flake8` complains about the non-usage of `vm` variable. I've tried to add something like `# noqa isort: split` and similar combinations, but only on of the tools (flake8 and isort) ignores the line, not both at at the same time...

Comment (jorgepiloto) [reply orekit/src/orekit_orbit.py]: Alright, everything should be fixed now. Although one inline comment is required, everything looks more "natural".

Comment (astrojuanlu) [reply orekit/src/orekit_orbit.py]: I see! If the `vm` variable is not used, perhaps `import utils` with a comment explaning what happens should be enough. However, it's good enough this way :)

**Review by astrojuanlu — APPROVED** [review]
This looks very good! Feel free to merge when the quality checks and the tests pass 🚀

**Comment by jorgepiloto** [comment]
Sorry for this mess with the quality checks. I am not sure why my local configuration does not match the one from the actions YAML file :sweat:

**Comment by jorgepiloto** [comment]
Merging this! All tests passing now :rocket:

**Pull request merged**